pkg/monitor/intervalcreation/node.go

```diff
@@ -100,3 +114,185 @@ func IntervalsFromEvents_NodeChanges(events monitorapi.Intervals, _ monitorapi.R
 
 	return intervals
 }
+
+func IntervalsFromNodeLogs(ctx context.Context, kubeClient kubernetes.Interface, beginning, end time.Time) (monitorapi.Intervals, error) {
+	ret := monitorapi.Intervals{}
+
+	allNodes, err := kubeClient.CoreV1().Nodes().List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	lock := sync.Mutex{}
+	errCh := make(chan error, len(allNodes.Items))
+	wg := sync.WaitGroup{}
+	for _, node := range allNodes.Items {
+		wg.Add(1)
+		go func(ctx context.Context, nodeName string) {
+			defer wg.Done()
+
+			// TODO limit by begin/end here instead of post-processing
+			nodeLogs, err := nodedetails.GetNodeLog(ctx, kubeClient, nodeName, "kubelet")
+			if err != nil {
+				errCh <- err
+				return
+			}
+			newEvents := eventsFromKubeletLogs(nodeName, nodeLogs)
+
+			lock.Lock()
+			defer lock.Unlock()
+			ret = append(ret, newEvents...)
+		}(ctx, node.Name)
+	}
+	wg.Wait()
+
+	errs := []error{}
+	for len(errCh) > 0 {
+		err := <-errCh
+		errs = append(errs, err)
+	}
+
+	return ret, utilerrors.NewAggregate(errs)
+}
+
+func eventsFromKubeletLogs(nodeName string, kubeletLog []byte) monitorapi.Intervals {
+	ret := monitorapi.Intervals{}
+
+	scanner := bufio.NewScanner(bytes.NewBuffer(kubeletLog))
+	for scanner.Scan() {
+		currLine := scanner.Text()
+		ret = append(ret, readinessFailure(currLine)...)
+		ret = append(ret, readinessError(currLine)...)
+	}
+
+	return ret
+}
+
+type kubeletLogLineEventCreator func(logLine string) monitorapi.Intervals
+
+var lineToEvents = []kubeletLogLineEventCreator{}
+
+func readinessFailure(logLine string) monitorapi.Intervals {
+	if !strings.Contains(logLine, `Probe failed`) {
+		return nil
+	}
+	if !strings.Contains(logLine, `probeType="Readiness"`) {
+		return nil
+	}
+
+	failureOutputRegex.MatchString(logLine)
+	if !failureOutputRegex.MatchString(logLine) {
+		return nil
+	}
+	outputSubmatches := failureOutputRegex.FindStringSubmatch(logLine)
+	message := outputSubmatches[1]
+	message, _ = strconv.Unquote(`"` + message + `"`)
```

Comment:
Could use a comment, looks like you're quoting and then unquoting. Also ignored error.